From b53bcd4089d115746c37c2de2df547650b437620 Mon Sep 17 00:00:00 2001
From: Alejandro Mery <amery@geeks.cl>
Date: Tue, 9 Oct 2012 17:49:43 +0200
Subject: [PATCH 783/944] arm: remove boot hacks from Allwinner

we don't need early (before iomap) access to dramc or script.bin, and we
assume the bootloader will pass the right machine type
---
 arch/arm/kernel/head.S | 47 -----------------------------------------------
 1 file changed, 47 deletions(-)

diff --git a/arch/arm/kernel/head.S b/arch/arm/kernel/head.S
index a982535..3bf0c7f 100644
--- a/arch/arm/kernel/head.S
+++ b/arch/arm/kernel/head.S
@@ -23,7 +23,6 @@
 #include <asm/thread_info.h>
 #include <asm/pgtable.h>
 
-#include <mach/entry-macro.S>
 #ifdef CONFIG_DEBUG_LL
 #include <mach/debug-macro.S>
 #endif
@@ -92,26 +91,6 @@ ENTRY(stext)
  THUMB(	.thumb			)	@ switch to Thumb now.
  THUMB(1:			)
 
-	mov	r0, #0
-	b	_real_start
-
-#if defined(CONFIG_ARCH_SUN3I)
-1:	.long 0x1094
-	.long 0x81000000
-#elif defined(CONFIG_ARCH_SUN4I)
-1:	.long 0x1008
-	.long 0x81000000
-#elif defined(CONFIG_ARCH_SUN5I)
-1:      .long 0x102A
-        .long 0x81000000
-#else
-#error "unsupport chip"
-#endif
-
-_real_start:
-	adr	r4, 1b
-	ldmia	r4, {r1, r2}
-
 	setmode	PSR_F_BIT | PSR_I_BIT | SVC_MODE, r9 @ ensure svc mode
 						@ and irqs disabled
 	mrc	p15, 0, r9, c0, c0		@ get processor id
@@ -247,7 +226,6 @@ __create_page_tables:
 	orr	r3, r7, r3, lsl #SECTION_SHIFT
 	add	r0, r4,  #(KERNEL_START & 0xff000000) >> (SECTION_SHIFT - PMD_ORDER)
 	str	r3, [r0, #((KERNEL_START & 0x00f00000) >> SECTION_SHIFT) << PMD_ORDER]!
-
 	ldr	r6, =(KERNEL_END - 1)
 	add	r0, r0, #1 << PMD_ORDER
 	add	r6, r4, r6, lsr #(SECTION_SHIFT - PMD_ORDER)
@@ -273,11 +251,6 @@ __create_page_tables:
 	bls	1b
 #endif
 
-	/* Add a map for script.bin */
-	add     r0, r4,  #((0x3000000 + KERNEL_START) & 0xff000000) >> 18
-	add     r6, r3, #48 << 20
-	str     r6, [r0, #((0x3000000 + KERNEL_START) & 0x00f00000) >> 18]
-
 	/*
 	 * Then map boot params address in r2 or the first 1MB (2MB with LPAE)
 	 * of ram if boot params address is not specified.
@@ -291,26 +264,6 @@ __create_page_tables:
 	orr	r6, r7, r0
 	str	r6, [r3]
 
-	/*
-	 * map io space(3MB is enough on our platform)
-	 *
-	 */
-	getioaddr r7, r3
-
-	mov	r3, r3, lsr #20
-	mov	r3, r3, lsl #2
-
-	add	r0, r4, r3
-	add	r6, r0, #0xc
-
-	mov	r3, r7, lsr #20
-	ldr	r7, [r10, #PROCINFO_IO_MMUFLAGS] @ io_mmuflags
-	orr	r3, r7, r3, lsl #20
-1:	str	r3, [r0], #4
-	add	r3, r3, #1 << 20
-	teq	r0, r6
-	bne	1b
-
 #ifdef CONFIG_DEBUG_LL
 #if !defined(CONFIG_DEBUG_ICEDCC) && !defined(CONFIG_DEBUG_SEMIHOSTING)
 	/*
-- 
1.8.0

